support/compressxdr/compress_xdr.go

```diff
@@ -3,36 +3,50 @@ package compressxdr
 import (
 	"io"
 
-	"github.com/stellar/go/support/errors"
+	xdr3 "github.com/stellar/go-xdr/xdr3"
 )
 
-const (
-	GZIP = "gzip"
-)
+func NewXDREncoder(compressor Compressor, xdrPayload interface{}) XDREncoder {
+	return XDREncoder{Compressor: compressor, XdrPayload: xdrPayload}
+}
+
+func NewXDRDecoder(compressor Compressor, xdrPayload interface{}) XDRDecoder {
+	return XDRDecoder{Compressor: compressor, XdrPayload: xdrPayload}
```

Comment:
```suggestion
func NewXDRDecoder(compressor Compressor, xdrPayload interface{}) *XDRDecoder {
	return &XDRDecoder{Compressor: compressor, XdrPayload: xdrPayload}
```

`NewXDRDecoder()` should return a pointer because `ReadFrom(r io.Reader)` modifies `XdrPayload`

Reply:
actually, I'm sorry I think `NewXDRDecoder()` should remain as is (return a value instead of a pointer). I didn't consider that `XdrPayload` is actually a pointer value. However, the `ReadFrom()` function on `XDRDecoder` should be a value receiver and not a pointer receiver